service/src/main/resources/api/service_openapi.yaml

```diff
@@ -1537,6 +1541,53 @@ components:
           items:
             $ref: "#/components/schemas/Filter"
 
+    OccurrenceForPrimaryFilter:
+      type: object
+      required: [entityGroup, occurrenceEntity]
+      description: |
+        Filters an occurrence entity with a criteria-occurrence entity group (e.g. condition
+        occurrences for people with gender=female, condition=diabetes).
+      properties:
+        entityGroup:
+          type: string
+        occurrenceEntity:
+          type: string
+        primarySubfilter:
+          $ref: "#/components/schemas/Filter"
+
+    PrimaryWithCriteriaFilter:
+      type: object
+      required: [entityGroup]
+      description: |
+        Filter the primary entity with a criteria-occurrence entity group (e.g. people where 
+        condition=diabetes and visit_type=inpatient).
+        If the group_by_count_operator is set, then the group_by_count_value must also be set.
+        The group_by_count_attributes are optional. If undefined, we group by the id only.
+        When group_by fields are set, we filter on # of related occurrence instances.
+        e.g. filter person instances with >2 related condition_occurrence instances
+      properties:
+        entityGroup:
+          type: string
+        criteriaSubfilter:
+          $ref: "#/components/schemas/Filter"
+        occurrenceSubfiltersAndGroupByAttributes:
```

Comment:
What is the purpose of this?

Reply:
It's for modifiers (e.g. visit type, age at occurrence) and group by attributes (e.g. start_date, date).

Reply:
Why is in a separate object though? I guess maybe the keys would be for each occurrence entity?

Reply:
Right, this allows for different sub-filters and group by attributes per occurrence entity. I'll add a description to make it clear the keys are the occurrence entity names.